src/improvements/tasks/StackedSimulator.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {Script} from "forge-std/Script.sol";
+import {TaskRunner} from "src/improvements/tasks/TaskRunner.sol";
+import {LibString} from "@solady/utils/LibString.sol";
+import {AccountAccessParser} from "src/libraries/AccountAccessParser.sol";
+import {console} from "forge-std/console.sol";
+import {VmSafe} from "forge-std/Vm.sol";
+
+/// This script enables stacked simulations. Stacked simulations allow us to simulate a task
+/// that depends on the state of another task that hasn't been executed yet.
+/// Only non-terminal tasks are executed as part of the stacked simulation.
+contract StackedSimulator is Script {
+    using LibString for string;
+    using AccountAccessParser for VmSafe.AccountAccess[];
+
+    struct TaskInfo {
+        string path;
+        string network;
+        string name;
+    }
+
+    function simulateStack(string memory network) public {
+        TaskInfo[] memory tasks = getNonTerminalTasks(network);
+        simulateStack(network, tasks[tasks.length - 1].name);
+    }
+
+    /// @notice Simulates the execution of a task and all tasks that must be executed before it.
+    /// This function copies the necessary files to a local directory that is not tracked by git.
+    /// It does this because as part of stacked simulations, we need to modify the task config.toml files
+    /// and we don't want to commit those changes to the repo.
+    function simulateStack(string memory network, string memory task) public {
```

Comment:
How do we coordinate ahead of time. We can add `[stateOverrides]`. But will require coordination.

Reply:
Sequence tasks of both styles in parallel.